### PR TITLE
Externalize node informers for node authz

### PR DESCRIPTION
**Pull request by yue9944882**
the pull will completely externalize node authz together with #67194

ref: #66680

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/auth/authorizer/node/graph.go [hunk 1]
```diff
@@ -318,7 +318,7 @@ func (g *Graph) AddPod(pod *corev1.Pod) {
 	// Short-circuit adding edges to other resources for mirror pods.
 	// A node must never be able to create a pod that grants them permissions on other API objects.
 	// The NodeRestriction admission plugin prevents creation of such pods, but short-circuiting here gives us defense in depth.
-	if _, isMirrorPod := pod.Annotations[api.MirrorPodAnnotationKey]; isMirrorPod {
```
Comment (yue9944882) [inline plugin/pkg/auth/authorizer/node/graph.go]: sadly, it should have been done in #67116. my mistake 🤕

File: pkg/kubeapiserver/authorizer/config.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func (config AuthorizationConfig) New() (authorizer.Authorizer, authorizer.RuleR
 			graph := node.NewGraph()
 			node.AddGraphEventHandlers(
 				graph,
-				config.InformerFactory.Core().InternalVersion().Nodes(),
+				config.VersionedInformerFactory.Core().V1().Nodes(),
 				config.VersionedInformerFactory.Core().V1().Pods(),
 				config.InformerFactory.Core().InternalVersion().PersistentVolumes(),
```
Comment (mikedanese) [inline pkg/kubeapiserver/authorizer/config.go]: What about this one?

Comment (yue9944882) [reply pkg/kubeapiserver/authorizer/config.go]: it on-going in #67194

Comment (mikedanese) [reply pkg/kubeapiserver/authorizer/config.go]: Great, thanks.

**Comment by deads2k** [comment]
looks good.  Once the prior one merges, this one can remove the old internal informers, right?

File: cmd/kube-apiserver/app/server.go [hunk 3]
```diff
@@ -526,7 +526,7 @@ func buildGenericConfig(
 		return
 	}
 
-	genericConfig.Authorization.Authorizer, genericConfig.RuleResolver, err = BuildAuthorizer(s, sharedInformers, versionedInformers)
```
Comment (yue9944882) [inline cmd/kube-apiserver/app/server.go]: > this one can remove the old internal informers, right?

for sure. it's complete gone from authz 🎉

**Comment by deads2k** [comment]
please squash to two commits

/lgtm
/hold
for squash

**Pull request merged**